### PR TITLE
aws-c-common/0.6.7

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **aws-c-common/0.6.7**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/aws-c-common/all/conanfile.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
-from conans import CMake, ConanFile, tools
 import os
+from conans import ConanFile, CMake, tools
 
-required_conan_version = ">=1.28.0"
+required_conan_version = ">=1.33.0"
```
Comment (SpaceIm) [inline recipes/aws-c-common/all/conanfile.py:4]: why?

File: recipes/aws-c-common/all/conanfile.py [hunk 2]
```diff
@@ -39,9 +39,8 @@ def configure(self):
         del self.settings.compiler.libcxx
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_folder = "aws-c-common-{}".format(self.version)
-        os.rename(extracted_folder, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+            destination=self._source_subfolder, strip_root=True)
```
Comment (SpaceIm) [inline recipes/aws-c-common/all/conanfile.py:43]: ahah, now you need conan 1.33.0

Comment (MartinDelille) [reply recipes/aws-c-common/all/conanfile.py:43]: Ok! 😅

File: recipes/aws-c-common/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -4,7 +4,5 @@ project(test_package LANGUAGES C)
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
 
-find_package(aws-c-common REQUIRED CONFIG)
-
 add_executable(${PROJECT_NAME} test_package.c)
-target_link_libraries(${PROJECT_NAME} AWS::aws-c-common)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (SpaceIm) [inline recipes/aws-c-common/all/test_package/CMakeLists.txt:8]: No please

Comment (MartinDelille) [reply recipes/aws-c-common/all/test_package/CMakeLists.txt]: ok

**Review by madebr — DISMISSED** [review]

**Comment by MartinDelille** [comment]
Why do I have this error? 

```
ERROR: 
	ConanException: aws-c-common/0.6.7 package_info(): self.cpp_info.components cannot be used with self.cpp_info global values at the same time
```

File: recipes/aws-c-common/all/conanfile.py [hunk 4]
```diff
@@ -51,23 +50,21 @@ def _configure_cmake(self):
...
     def package(self):
         self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
         cmake = self._configure_cmake()
         cmake.install()
-
         tools.rmdir(os.path.join(self.package_folder, "lib", "aws-c-common"))
 
     def package_info(self):
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs = ["dl"]
```
Comment (madebr) [inline recipes/aws-c-common/all/conanfile.py:67]: You cannot combine global properties `self.cpp.info.system_libs` with usage of components (`self.cpp_info.components`).

A few lines later in this recipe, there is already:
```python
        if self.settings.os == "Linux":
            self.cpp_info.components["aws-c-common-lib"].system_libs = ["m", "pthread", "rt"]
```
So is this addition needed?

Comment (MartinDelille) [reply recipes/aws-c-common/all/conanfile.py]: Ok my bad. Fixed here: c200e16

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**